client/src/components/modal/modal.js

```diff
@@ -0,0 +1,21 @@
+import ko from 'knockout';
+
+import './modal.less';
+import template from 'components/modal/modal.html';
+import register from 'components/component-registrator';
+
+class ModalViewModel {
+    constructor(params) {
+        this.comp = params.comp || {};
+        this.data = params.data || {};
+        this.isVisible = params.visible || ko.observable(false);
+        this.parentComponent = params.parentComponent;
+    }
+    hide(viewModel, event) {
+        if (event.target.className.indexOf('modal--visible') !== -1) {
```

Comment:
It really seems like `this.isVisible` should have this value. Checking class name is REALLY bad.